# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/VideoFrameContainer.h
# Commit: 35d86243cb45
# Full Hash: 35d86243cb45f7e02dd6ec0eea04cd62c2a70196
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2024-09-16 09:14:26
# Regressor Bug: 1908245
# File Overlap Count: 1
# Description:
#   Bug 1908245 - Add support for missing requestVideoFrameCallback parameters. r=media-playback-reviewers,webidl,padenot,saschanaz
#   
#   This patch adds support for exposing the capture time, receive time,
#   processing duration and RTP timestamp parameters on the callback for
#   rVFC for WebRTC related elements.
# ==============================================================================

diff -r 7d143d92bf2a -r 35d86243cb45 dom/media/VideoFrameContainer.h
--- a/dom/media/VideoFrameContainer.h	Sun Sep 15 20:03:40 2024 +0000
+++ b/dom/media/VideoFrameContainer.h	Sun Sep 15 21:49:38 2024 +0000
@@ -41,7 +41,9 @@
                       already_AddRefed<ImageContainer> aContainer);
 
   void SetCurrentFrame(const gfx::IntSize& aIntrinsicSize, Image* aImage,
-                       const TimeStamp& aTargetTime);
+                       const TimeStamp& aTargetTime,
+                       const media::TimeUnit& aProcessingDuration,
+                       const media::TimeUnit& aMediaTime);
   // Returns the last principalHandle we notified mElement about.
   PrincipalHandle GetLastPrincipalHandle();
   PrincipalHandle GetLastPrincipalHandleLocked() MOZ_REQUIRES(mMutex);